ENIGMAsystem/SHELL/Widget_Systems/General/tinyfiledialogs/dialogs.cpp

```diff
@@ -317,6 +349,7 @@ namespace enigma_user
   {
     fname = tfd_add_escaping(fname);
     filter = tfd_add_escaping(filter);
+    filter = detect_all_files_filter(filter);
```

Comment:
Having four copies of the same logic requires you to make the same modification in four places every time. On that note, are you sure you've made this modification everywhere it's required?

Reply:
Well, good job, I am glad he used a helper for this, he's definitely improving.